guides/v2.2/coding-standards/technical-guidelines.md

```diff
@@ -455,6 +455,8 @@ class View extends Template
 
 4.4. Plugins MUST be stateless.
 
+4.5. Plugins SHOULD NOT change a state of an intercepted object.
```

Comment:
Should we clarify 4.3 instead of adding new item?

Reply:
I believe it is a little different case.
We do not want to support plugins for all DTO getters and setters.
Service method can be pluginized. but the plugin should not change an execution context. Invoking transaction, changing app state, etc.

Reply:
Invoking transaction doesn't change state of intercepted object or does it? If we can call a method that it's not setter (otherwise it will be 4.3) to change application state, how plugin is different from other methods that can be used to call this method?

Reply:
For example, we have an extension which introduces a plugin for some collection method. Inside of this plugin,  by using the collection's public method `setItemObjectClass` a developer can redefine the item type. Such a change will be destructive to all collections clients who expect items of the defined type.

Reply:
We should provide specific examples. Even though in general it makes sense to restrict modification of the intercepted object, there still might be situations when it is useful.

Reply:
"SHOULD" here covers some situations.